arches/app/views/search.py

```diff
@@ -205,6 +213,8 @@ def search_terms(request):
         query.add_aggregation(base_agg)
 
         ret[index] = []
+        if len(permitted_nodegroups) == 0:
+            continue
```

Comment:
Won't this short-circuit a concepts search when it shouldn't? If it's intentional, let's move it outside the loop to clarify.

Reply:
I agree - if a user doesn't have access to any nodegroups, then it seems like this should be moved out of the loop.

Reply:
I think we should check this first thing and if it's empty immediately return an empty array to the front end

Reply:
change in latest